devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy

```diff
@@ -0,0 +1,683 @@
+}
+
+private getEventMap(name, value, displayed=null, desc=null, unit=null) {
+	def isStateChange = (device.currentValue(name) != value)
+	displayed = (displayed == null ? isStateChange : displayed)
+	def eventMap = [
+			name: name,
+			value: value,
+			displayed: displayed,
+			isStateChange: isStateChange,
+			descriptionText: desc ?: "${device.displayName} ${name} is ${value}"
+	]
+
+	if (unit) {
+		eventMap.unit = unit
+		eventMap.descriptionText = "${eventMap.descriptionText}${unit}"
+	}
+	if (displayed) {
+		logDebug "${eventMap.descriptionText}"
+	}
+	return eventMap
+}
+
+
+private wakeUpNoMoreInfoCmd() {
+	return secureCmd(zwave.wakeUpV1.wakeUpNoMoreInformation())
+}
+
+private batteryGetCmd() {
+	return secureCmd(zwave.batteryV1.batteryGet())
+}
+
+private sensorBinaryGetCmd() {
+	return secureCmd(zwave.sensorBinaryV2.sensorBinaryGet())
+}
+
+private sensorMultilevelGetCmd(sensorType) {
+	def scale = (sensorType == tempSensorType) ? 0 : 1
+	return secureCmd(zwave.sensorMultilevelV5.sensorMultilevelGet(scale: scale, sensorType: sensorType))
+}
+
+private configGetCmd(param) {
+	return secureCmd(zwave.configurationV1.configurationGet(parameterNumber: param.num))
+}
+
+private configSetCmd(param, value) {
+	return secureCmd(zwave.configurationV1.configurationSet(parameterNumber: param.num, size: param.size, scaledConfigurationValue: value))
+}
+
+private secureCmd(cmd) {
+	try {
+		if (zwaveInfo?.zw?.contains("s") || ("0x98" in device?.rawDescription?.split(" "))) {
+			return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		}
+		else {
+			return cmd.format()
+		}
+	}catch (ex) {
+		return cmd.format()
+	}
+}
+
+
+private static getCommandClassVersions() {
+	[
+			0x30: 2,  // SensorBinary
+			0x31: 5,  // SensorMultilevel
+			0x55: 1,  // TransportServices
+			0x59: 1,  // AssociationGrpInfo
+			0x5A: 1,  // DeviceResetLocally
+			0x5E: 2,  // ZwaveplusInfo
+			0x6C: 1,  // Supervision
+			0x70: 1,  // Configuration
+			0x71: 3,  // Notification
+			0x72: 2,  // ManufacturerSpecific
+			0x73: 1,  // Powerlevel
+			0x7A: 2,  // FirmwareUpdateMd
+			0x80: 1,  // Battery
+			0x84: 1,  // WakeUp
+			0x85: 2,  // Association
+			0x86: 1,  // Version
+			0x8E: 2,  // MultChannelAssociation
+			0x87: 1,  // Indicator
+			0x9F: 1   // Security 2
+	]
+}
+
+
+private canReportBattery() {
+	return state.refreshSensors || (!isDuplicateCommand(state.lastBattery, (12 * 60 * 60 * 1000)))
+}
+
+private getPendingChanges() {
+	return configParams.count { "${it.value}" != "${getParamStoredValue(it.num)}" }
+}
+
+private getParamStoredValue(paramNum) {
+	return safeToInt(state["configParam${paramNum}"] , null)
+}
+
+private setParamStoredValue(paramNum, value) {
+	state["configParam${paramNum}"] = value
+}
+
+
+// Sensor Types
+private static getTempSensorType() { return 1 }
+private static getLightSensorType() { return 5 }
+
+
+// Configuration Parameters
+private getConfigParams() {
+	[
+			batteryReportThresholdParam,
```

Comment:
Could You indent this section with 2 tabs ?

Reply:
OK, I adjust it.